Program.cs

```diff
@@ -32,11 +34,19 @@ private static void Main(string[] args)
         app.MapGet("/person/GetByNameAndType", (PersonService personService, [FromQuery] string name, string emplType) 
         => Results.Ok(personService.GetPersonByNameAndPersonType(name, emplType)));
         //Products methods
+        
         app.MapGet("/product", (ProductService productService) => Results.Ok(productService.GetAll()));
         app.MapGet("/product/GetByName", (ProductService productService, [FromQuery] string name) => Results.Ok(productService.GetProductsByName(name)));
         app.MapGet("/product/GetByCatType", (ProductService productService, [FromQuery] string catType) => Results.Ok(productService.GetProductByCategoryType(catType)));
         app.MapGet("/product/GetByNameAndType", (ProductService productService, [FromQuery] string name, string catType) 
         => Results.Ok(productService.GetProductByNameAndCategoryType(name, catType)));
+        
+
+        //Sales methods
+        app.MapGet("/SalesTest",() => "Esta es una pruba");
+        app.MapGet("/sale",(SaleService salesService) => Results.Ok(salesService.GetAll()));
+        app.MapGet("/sale/GetByNameAndYear",(SaleService saleService,[FromQuery] string name, string yearSale) => Results.Ok(saleService.GetBynameAndYear(name,yearSale)));
```

Comment:
This method is well-defined, but as part of the challenge, it's crucial to determine whether certain information, especially critical business information, should be included. It would be beneficial to add the 'orderDate' to this JSON response. This way, the information can be tracked over time, providing a means to search for order details effectively. As it stands, while the method is functional, the information provided is not particularly useful for tracking orders.